Dockerfile

```diff
@@ -1,68 +1,28 @@
-FROM peenuty/rails-passenger-nginx-docker-i:1.0.1
+FROM golang:1.9
 MAINTAINER Mike Rapadas <mike@mrap.me>
 
-# Install Node
+COPY . /go/src/github.com/mrap/combo
 
-    # GET NODE INSTALL DEPS
-    RUN       apt-get update --fix-missing
-    RUN       apt-get install -y build-essential curl libssl-dev
+WORKDIR /go/src/github.com/mrap/combo
 
-    # INSTALL NODE
-    RUN       curl -sL https://deb.nodesource.com/setup_6.x | bash
-    RUN       apt-get install nodejs
+RUN curl -sL https://deb.nodesource.com/setup_6.x | bash && \
+  curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+  echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
+  apt-get update && \
+  apt-get install -y nodejs yarn && \
+  \
+  yarn install && \
+  yarn global add bower grunt-cli && \
+  bower install --allow-root && \
+  grunt && \
+  \
+  go get github.com/gin-gonic/gin && \
+  go get github.com/michelloworld/ez-gin-template && \
+  go get github.com/mrap/stringutil && \
+  go get github.com/mrap/wordpatterns
 
-# Grunt needs git
-    RUN apt-get -y install git
-
-# Install Sass
-
-    RUN bash -l -c "gem install sass"
-
-# Install grunt
-    RUN npm install -g grunt-cli
-
-# Install Bower
-    RUN npm install -g bower
-
-
-# Setup Go
-#
-# Go Dockerfile
-#
-# https://github.com/dockerfile/go
-#
-
-# Install Go
-RUN \
-  mkdir -p /goroot && \
-    curl https://storage.googleapis.com/golang/go1.7.linux-amd64.tar.gz | tar xvzf - -C /goroot --strip-components=1
-
-# Set environment variables.
-ENV GOROOT /goroot
-ENV GOPATH /gopath
-ENV PATH $GOROOT/bin:$GOPATH/bin:$PATH
-
-# Define working directory.
-ADD . /gopath/src/combo
-WORKDIR /gopath/src/combo
-
-# Install Go dependencies
-RUN apt-get install -y mercurial
-RUN go get github.com/mrap/wordpatterns
-RUN go get github.com/mrap/stringutil
-RUN go get github.com/revel/revel
-
-# Prepare frontend files
-RUN npm install
-# manually npm install grunt-sass to ensure libsass bindings
-RUN npm install grunt-sass
-RUN bower --allow-root install
-RUN grunt
-
-# Install Revel CMD
-RUN go get github.com/revel/cmd/revel
-
-# Start the app
-CMD revel run combo prod
+WORKDIR /go/src/github.com/mrap/combo/functions/api
 
 EXPOSE 9000
+
+CMD ["go", "run", "./serve/main.go"]
```

Comment:
`RUN` and `CMD` commands should be the same format. I personally favor the non-array syntax since it looks like plain ol' shell, but let's pick one and stick with it for consistency and readability :)

Reply:
Best practices for `CMD` is to use the array syntax according to the [Docker Docs](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#cmd).

But I am open to using the shell syntax for this project, I agree that it is more readable.